### PR TITLE
docs(integrate): Google Cloud with Workforce Identity Federation (OIDC)

### PR DESCRIPTION
**Pull request by mffap**
Adds an integration guide for Google Cloud with Workforce Identity Federation (OIDC) using ZITADEL as external identity provider. This guide also covers the required custom claims with Actions.

```[tasklist]

### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

File: docs/docs/guides/integrate/services/google-cloud.mdx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+:::info
+Google Cloud requires just an ID Token as JWT including the [described required and optional scopes](https://cloud.google.com/iam/docs/workforce-identity-federation#attribute-mappings).
+:::
+
+Create a new application and click on "I'm a pro. Skip this wizard."
+
+- **Application Type**: Web
+- **Grant Types**: Authorization Code, Implicit, Refresh Token
+- **Response Type**: Code, ID Token, Token-ID Token
```
Comment (livio-a) [inline docs/docs/guides/integrate/services/google-cloud.mdx:46]: probably `Implicit` and `ID Token` would be sufficient... do we have the possibility to test it ourselves?

File: docs/docs/guides/integrate/services/google-cloud.mdx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+:::
+
+## **ZITADEL**: Token settings
+
+![Token settings](/img/guides/integrate/services/google-cloud-token-settings.png)
+
+After creating, go to the application settings "Token settings" and configure as follows:
+
+- **Auth Token Type**: JWT
+- **Add user roles to the access token**: enabled
```
Comment (livio-a) [inline docs/docs/guides/integrate/services/google-cloud.mdx:75]: as above... if only id token is used, access token type and roles won't be necessary

File: docs/docs/guides/integrate/services/google-cloud.mdx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  let grants = [];
+  ctx.v1.user.grants.grants.forEach(claim => {
+    claim.roles.forEach(role => {
+        grants.push(claim.projectId+':'+role)  
+    })
+  })
+
+  api.v1.claims.setClaim('google.groups', grants)
+  api.v1.claims.setClaim('google.display_name', ctx.v1.user.human.displayName)
+}
```
[the remark this answers is not in the document]
Comment (livio-a) [reply docs/docs/guides/integrate/services/google-cloud.mdx:110]: we fixed the action in the meantime

**Review by livio-a — CHANGES_REQUESTED** [review]
sorry... didn't notice this PR
we did a fix in the action and i have some questions regarding the app setup

File: docs/docs/guides/integrate/services/google-cloud.mdx (hunk 1 above)
Comment (mffap) [reply docs/docs/guides/integrate/services/google-cloud.mdx:46]:
```suggestion
- **Grant Types**: Authorization Code, Implicit, Refresh Token
- **Response Type**: ID Token
```

File: docs/docs/guides/integrate/services/google-cloud.mdx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+First of all we create the application in your project.
+
+:::info
+Google Cloud requires just an ID Token as JWT including the [described required and optional scopes](https://cloud.google.com/iam/docs/workforce-identity-federation#attribute-mappings).
+:::
+
+Create a new application and click on "I'm a pro. Skip this wizard."
+
+- **Application Type**: Web
+- **Grant Types**: Authorization Code, Implicit, Refresh Token
```
Comment (mffap) [inline docs/docs/guides/integrate/services/google-cloud.mdx:45]:
```suggestion
- **Grant Types**: Implicit
```

File: docs/docs/guides/integrate/services/google-cloud.mdx (hunk 2 above)
Comment (mffap) [inline docs/docs/guides/integrate/services/google-cloud.mdx:75]:
```suggestion
- **Add user roles to the access token**: disabled (optional)
```

**Review by livio-a — APPROVED** [review]

**Pull request merged**